capstone/src/main/webapp/scripts/poll.js

```diff
@@ -26,16 +27,49 @@ function getPollOptions() {
       const optionsContainer = document.getElementById("options-container");
       optionsContainer.innerHTML = "";
       let maxVotes = getMaxVotes(pollData);
+      topChallenge = pollData["options"][0]["text"];
       pollData["options"].forEach(option => {
         renderOptionElement(option, maxVotes);
       });
       return pollData["votedOptions"];
     })
     .then(votedOptions => {
       handleCheck(votedOptions);
+    })
+    .then(checkWeek);
+}
+
+function checkWeek() {
+  getChallenge();
+  let now = new Date();
+  //Hard coding due date for now
+  let millisTillDueDate =
+    new Date(now.getFullYear(), now.getMonth(), now.getDate(), 13, 04, 0, 0) -
+    now;
```

Comment:
I know you are just hardcoding the value for now, but can you explain to me the logic for the future? Importantly, where are you going to get the dueDate from? Will it be from the getCallenge() which just currently doesn't return a challenge?

Reply:
I've written the logic in the new pull request. I get the due date from the fetch request in getChallenge, and this due date is passed from getChallenge into checkWeek()

Reply:
Okie dokie thanks for letting me know!